cmd/argo-watcher/argo_api_test.go

```diff
@@ -72,7 +73,8 @@ func init() {
 	mux.HandleFunc("/api/v1/session/userinfo", getUserInfoHandler)
 	mux.HandleFunc("/api/v1/applications/test", getApplicationHandler)
 	server = httptest.NewServer(mux)
-	api = &ArgoApi{baseUrl: server.URL, client: server.Client()}
+	parsedURL, _ := url.Parse(server.URL) // we assume that the server.URL is valid
```

Comment:
We might want to not assume, but throw errors here

Reply:
It is a URL coming from the default library, If it will return an invalid `url` it would be the least of our problems.